annotation/tests/test_annotations.py

```diff
@@ -235,3 +274,156 @@ def test_create_manifest_json(
             annotated_doc.tenant,
             moto_s3,
         )
+
+
+@pytest.mark.parametrize(
+    ("is_latest", "new_equal_latest"),
+    ((True, False), (True, True)),  # Success path  # return latest_doc
+)
+def test_construct_annotated_doc(
+    annotated_doc,
+    annotation_doc_for_save,
+    is_latest,
+    new_equal_latest,
+):
+    doc = annotation_doc_for_save
+    latest_doc = annotated_doc
+    s3_path = (
+        f"{S3_START_PATH}/{annotated_doc.job_id}/" f"{annotated_doc.file_id}"
+    )
+    s3_file_path = "path"
+    s3_file_bucket = "bucket"
+
+    pages_sha_latest = ({"1": "a"}, "b")
+    pages_sha_non_latest = ({"1": "a"}, "c")
+
+    db = Mock()
+
+    with patch.object(db, "add") as mock_db_add, patch.object(
+        db, "add_all"
+    ) as mock_db_add_all, patch.object(
+        db,
+        "commit",
```

Comment:
You can add methods in db  mock instead of using patch several times.

Reply:
```suggestion
    db.add() = Mock()
    db.add_all() = Mock()
```
Would you say this is more readable? Should we do this for every db method?